runtime/ast/expression.go

```diff
@@ -769,6 +1019,28 @@ func (e *BinaryExpression) String() string {
 	)
 }
 
+func (e *BinaryExpression) Doc() prettier.Doc {
+	// TODO: potentially parenthesize
+	leftDoc := e.Left.Doc()
+
+	// TODO: potentially parenthesize
+	rightDoc := e.Right.Doc()
+
+	return prettier.Group{
+		Doc: prettier.Concat{
+			prettier.Group{
+				Doc: leftDoc,
+			},
+			prettier.Line{},
```

Comment:
`prettier.Line` is rendered as ae space if there is space, or a newline if there is not.

The idea here is to break before the symbol, so that e.g. `let a = 1 + 2 * 3` is rendered like this if it does not fit:
```
let a = 1
    + 2
    * 3
```